docs/fsharp/style-guide/formatting.md

```diff
@@ -1040,210 +852,408 @@ with
 
 ```fsharp
-functionName 
-    arg1 
-    arg2 
-    arg3 
-    (fun arg4 ->
-        bodyExpr)
+// OK
+type PostalAddress =
+    { Address: string
+      City: string
+      Zip: string }
+    member x.ZipAndCity = $"{x.Zip} {x.City}"
+```
 
-functionName 
-    arg1 
-    arg2 
-    arg3 
-    (function
-     | Choice1of2 x
-     | Choice2of2 y)
+Do not place the `{` at the end of the type declaration line, and do not use `with`/`end` for members, which are redundant.
+
+```fsharp
+// Not OK, code formatters will reformat to the above by default
+type PostalAddress = {
+    Address: string
+    City: string
+    Zip: string 
+  }
+  with
+    member x.ZipAndCity = $"{x.Zip} {x.City}"
+  end
 ```
 
-If the body of a lambda expression is multiple lines long, you should consider refactoring it into a locally-scoped function.
+When XML documentation is added for record fields, it becomes normal to indent and add whitespace:
+
+```fsharp
+type PostalAddress =
+    {
+        /// The address
+        Address: string
 
-Note that in combination of infix operators the body of a lambda expression should be indented one indent further from the current line.
-This is not the case when all arguments are indented one from the function application, there the indent should be in respect to the function name.
+        /// The city
+        City: string
+
+        /// The zip code
+        Zip: string
+    }
+
+    /// Format the zip code and the city
+    member x.ZipAndCity = $"{x.Zip} {x.City}"
+```
+
+Placing the opening token on a new line and the closing token on a new line is preferable if you are declaring interface implementations or members on the record:
 
 ```fsharp
-// With 4 spaces indentation
-list1
-|> List.iter (fun elem ->
-    // one indent starting from the pipe
-    printfn $"A very long line to format the value: %d{elem}")
+// Declaring additional members on PostalAddress
+type PostalAddress =
+    {
+        /// The address
+        Address: string
 
-list1
-|> List.fold
-       // one indent from the function name 
-       someLongParam
-       anotherLongParam
+        /// The city
+        City: string
 
-// With 2 spaces indentation
-list1
-|> List.fold
-     // one indent from the function name 
-     someLongParam
-     anotherLongParam
+        /// The zip code
+        Zip: string
+    }
+    member x.ZipAndCity = $"{x.Zip} {x.City}"
 
-list1
-|> List.iter (fun elem ->
-  // one indent starting from the pipe
-  printfn $"A very long line to format the value: %d{elem}")
+type MyRecord =
+    {
+        /// The record field
+        SomeField: int
+    }
+    interface IMyInterface
 ```
 
-When the function take a single multiline tuple argument, the same rules for [Formatting constructors, static members, and member invocations](#formatting-constructors-static-members-and-member-invocations) apply.
+## Formatting discriminated union declarations
```

Comment:
Should this be an H3 since it falls under broader "Formatting declarations" section?